shunting-yard.cpp

```diff
@@ -329,7 +329,7 @@ TokenQueue_t calculator::toRPN(const char* expr,
       // add the parsed number to the output queue.
       std::string key = rpnBuilder::parseVar(expr, &expr);
 
-      if ((parser=config.parserMap.find(key))) {
+      if (NULL != (parser = config.parserMap.find(key))) {
```

Comment:
Minor nitpick: I always feel strange when I see if-conditions organized with a constant on the left side. In spoken language, we usually ask questions in the form of: "Are you 22 yo?" so "you" comes before "22", we don't usually ask "is 22 years your age?", so when reading statements like that it confuses me a little bit.

Another example: "is the number of students in this class 40?" and not: "is 40 the number of students in this class?"

But since I see a lot of people doing it like this I imagine that there must be a reason behind this that I am not aware of.

So this is just a nitpicking suggestion you can keep like this if you prefer. Also if you have an argument about this I am interested in it hearing it =].

And again thanks for the contribution, I will finish this review tonight.

Reply:
Ah, Yoda conditions! It's an acquired habit, I admit.

The most commonly stated reason for these, and why some projects actually require them, is that it's easy to write accidental assignments like `if (age = 42)` if the left side of the comparison is an lvalue. with a constant on the left side, like `if (42 == age)`, accidental assignments (42 = age) cause a compiler error. 

The extra comparison here is added, because at high warning levels, MSVC will warn about assignments in conditions, in an attempt to mitigate the same sort of accident, and adding an explicit comparison to NULL signals that this is not an accident.

Another way I've seen this written is `(!!(parser = foo()))`, but the double exclamation mark is too easy to misread, IMHO.